src/main/java/com/micetr0/mock_DB/MySqlDB.java

```diff
@@ -258,38 +259,106 @@ public Boolean execute(Connection conn) throws SQLException {
     @Override
-    public List<Note> findNotesByMeasureIdAndMeasureIndex(Integer measureId, Integer measureIndex) {
+    public List<Composition> findCompositionsByAccountId(Integer accountId) {
         return null;
     }
 
     @Override
-    public List<Composition> findCompositionsIdsByAccountId(Integer accountId) {
+    public List<Note> findNotesByMeasureIdAndMeasureIndex(Integer measureId, Integer measureIndex) {
         return null;
     }
 
     @Override
     public List<Account> findAllAccounts() {
-        return null;
+        return executeTransaction(conn -> {
+            PreparedStatement getAccsStmt = null;
+            ResultSet accResultSet = null;
+
+            try{
+                getAccsStmt = conn.prepareStatement(
+                    "Select accounts.* from accounts"
+                );
+
+                List<Account> resultAccounts = new ArrayList<Account>();
+
+                accResultSet = getAccsStmt.executeQuery();
+
+                //for testing that result was returned i.e. accounts exist
+                Boolean found = false;
+
+                while(accResultSet.next()){
+                    found = true;
+
+                    Account account = new Account();
+                    loadAccount(account, accResultSet, 1);
+
+                    resultAccounts.add(account);
+                }
+
+                if(!found){
+                    System.out.println("No Account was found that matched that username");
+                }
+
+                return resultAccounts;
+            }
+            finally{
+                DBUtil.closeQuietly(getAccsStmt);
+                DBUtil.closeQuietly(accResultSet);
+            }
+        });
     }
 
     @Override
-    public List<Account> findCurrentAccount(Integer accountId) {
-        return null;
+    public List<Account> findAccountByAccountID(Integer accountId) {
+        return executeTransaction(conn -> {
+            PreparedStatement getAccStmt = null;
+            ResultSet accResultSet = null;
+
+            try{
+                getAccStmt = conn.prepareStatement(
+                        "Select accounts.* from accounts"+
+                                "where accountID = ?"
+                );
+                getAccStmt.setInt(1,accountId);
+
+                List<Account> resultAccounts = new ArrayList<>();
+
+                accResultSet = getAccStmt.executeQuery();
+
+                //for testing that result was returned i.e. accounts exist
+                Boolean found = false;
+
+                while(accResultSet.next()){
+                    found = true;
+
+                    Account account = new Account();
+                    loadAccount(account, accResultSet, 1);
+
+                    resultAccounts.add(account);
+                }
+
+                if(!found){
+                    System.out.println("No Account was found that matched that username");
+                }
+
+                return resultAccounts;
+            }
```

Comment:
Add a catch block for exceptions.
try -> catch - > finally

Reply:
nothing to catch here, no catches were implemented in Hake's code for the example ORM lab